libsql-server/src/namespace/mod.rs

```diff
@@ -645,9 +647,11 @@ impl<T: Database> Namespace<T> {
         Ok(())
     }
 
-    async fn shutdown(mut self) -> anyhow::Result<()> {
+    async fn shutdown(mut self, should_checkpoint: bool) -> anyhow::Result<()> {
         self.tasks.shutdown().await;
-        self.checkpoint().await?;
+        if should_checkpoint {
+            self.checkpoint().await?;
+        }
         self.db.shutdown().await?;
```

Comment:
I want to confirm one thing, I haven't changed `db::shutdown`, it has `replicator.wait_until_snapshotted().await?` This should be fine, right? As I understand, this one does not trigger a new upload rather wait for existing one to finish.

Reply:
@Horusiath ?

Reply:
Yes. When a new generation is made, we begin uploading a DB snapshot found at the beginning of that generation. `replicator.wait_until_snapshotted` is used to wait for that process to finish. This way we have an entire generation lifetime for it to complete which may be necessary since snapshot file may be massive.

Calling `checkpoint` twice in a row - once explicitly and once as part of db.shutdown - is basically a roundabout way to compress WAL into db file, trigger snapshot upload and wait for it.